tests/Unit/Shared/Infrastructure/DoctrineType/MongoDBDomainUuidTypeTest.php

```
@@ -0,0 +1,168 @@
+        );
+
+        $phpValue = $this->domainUuidType->convertToPHPValue($uuid);
+
+        $this->assertSame($uuid, $phpValue);
+    }
+
+    public function testConvertToDatabaseValueWithCustomUuidInterface(): void
+    {
+        $invalidUuidString = 'not-a-valid-uuid-format';
+        $customUuid = new class($invalidUuidString) implements UuidInterface {
+            public function __construct(private readonly string $value)
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$value'. (undefined)

(UnusedFormalParameter)